src/ns_tracker/nsdeps.clj

```diff
@@ -21,8 +22,42 @@
 	     (contains? #{:use :require} (first form)))
     (apply union (map #(deps-from-libspec nil %) (rest form)))))
 
+(defn- find-resource [path dirs]
+  (->> (map #(File. ^File % ^String path) dirs)
+       (filter #(.isFile ^File %))
+       first))
+
+(defn- print-warnings-for-missing-resources [namespace resources dirs]
+  (doseq [resource (->> resources
+                        (filter #(nil? (second %)))
+                        (map first))]
+    (binding [*out* *err*]
+      (let [dirs (vec (map #(.getPath ^File %) dirs))]
+        (println (str "ns-tracker: Unable to track dependency from namespace " namespace
+                      " to resource \"" resource "\". The resource was not found in directories " dirs "."))))))
+
+(defn- ns-attr-map [[_ _ docstring? attr-map?]]
+  (cond
+    (map? docstring?) docstring?
+    (map? attr-map?) attr-map?))
+
+(defn- ns-metadata [[_ name :as decl]]
+  (or (meta name)
```

Comment:
The `or` should be changed to `merge` to replicate the inbuild functionality of `ns`:

```clojure
user=> (ns ^{:a 1 :b 2} foo {:b 3 :c 4})
nil
foo=> (meta *ns*)
{:a 1, :b 3, :c 4}
```